### ISSUE TITLE
[JENKINS-37803] Show log before any flow nodes arrive

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#3166]
<p>It would be nice to be able to show the log for a pipeline job while it is waiting to run (ie it is running, but no flow nodes have arrived yet). </p>

<p>This would show up that it may be waiting on resources, or there is some imminent network failure. Once the flow nodes start arriving, it would need to be dismissed. </p>

<p>Needs some design thought... </p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale">michaelneale</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-37803" target="_blank">Show log before any flow nodes arrive</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay">jamesdumay</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: blueocean-plugin
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2016-09-19T01:57:33+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>It would be nice to be able to show the log for a pipeline job while it is waiting to run (ie it is running, but no flow nodes have arrived yet). </p>

<p>This would show up that it may be waiting on resources, or there is some imminent network failure. Once the flow nodes start arriving, it would need to be dismissed. </p>

<p>Needs some design thought... </p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-37803?focusedId=268486&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-268486">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>cc <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a></p>
</pre>
</details>

</ul>
</details>

<p>cc <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> tscherler:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-37803?focusedId=268520&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-268520">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>+1</p>
</pre>
</details>

</ul>
</details>

<p>+1</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-37803?focusedId=268580&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-268580">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'll see if James and I can cook up a design. <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a> the "easter egg" means this is available right, we just need to work out when/how to present it? Does it include tailing the log? </p>
</pre>
</details>

</ul>
</details>

<p>I'll see if James and I can cook up a design. <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a> the "easter egg" means this is available right, we just need to work out when/how to present it? Does it include tailing the log? </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> tscherler:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-37803?focusedId=268616&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-268616">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://ci.blueocean.io/blue/organizations/jenkins/Acceptance-Tests/detail/Acceptance-Tests/218/pipeline/?view=0" class="external-link" target="_blank" rel="nofollow noopener">https://ci.blueocean.io/blue/organizations/jenkins/Acceptance-Tests/detail/Acceptance-Tests/218/pipeline/?view=0</a> yes it is in and it did acted exactly as freestyle does, so yes it does karaoke.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://ci.blueocean.io/blue/organizations/jenkins/Acceptance-Tests/detail/Acceptance-Tests/218/pipeline/?view=0" class="external-link" target="_blank" rel="nofollow noopener">https://ci.blueocean.io/blue/organizations/jenkins/Acceptance-Tests/detail/Acceptance-Tests/218/pipeline/?view=0</a> yes it is in and it did acted exactly as freestyle does, so yes it does karaoke.</p>
